src/io.jl

```diff
@@ -81,6 +81,15 @@ function create_parameters_and_sets_from_file(input_folder::AbstractString)
     return params, sets
 end
 
+"""
+    read_csv_into_dataframe(csv_name)
+
+Reads the csv with 2 header rows and returns the dataframe
+"""
+function read_csv_into_dataframe(csv_name)
+    CSV.read(csv_name, DataFrames.DataFrame; header = 2)
```

Comment:
This is where you should validate the dataframe.  Otherwise this function is kinda pointless it replaces a one function call with another.  I would expect it to go something like this:

```
function get_df(path, dataset_type)
    df = CSV.read(...)
    validate_df(df, dataset_type, path)
    return df
end
```
`dataset_type` are the tuples we created in `input_tables.jl`

Reply:
Maybe for simplicity, `input_tables.jl` could also be merged with `io.jl`.  But that's only my personal preference.

Reply:
That would make sense since io stands for input/output right?

Reply:
Also thanks for showing how the validation should be added. I recognized that it was just replacing one function with another but wasn't sure what else the issue could mean.

Reply:
@suvayu Can you explain more what you mean by the argument dataset_type? I assume it has something to do with schema::DataType in the validate_df arguments, but I couldn't find a use of validate_df to see what that argument is in practice. Something to do with the structs earlier in the input_tables file?

Reply:
The idea is, we provide an expected schema, and a dataframe, and then compare if they are compatible.  Now schemas can be defined in numerous ways, but in the issue we worked together, we chose to do it as `struct`s.  If you were to instantiate a struct (as in, a struct with specific values for all the attributes), it would be the same as a valid row of the corresponding dataframe.  Since `struct`s are `DataType`s, that's why the argument to `validate_df` is `schema::DataType`.

As for an example, you can always look at the tests to see how to use a particular function: https://github.com/TulipaEnergy/TulipaEnergyModel.jl/blob/ad1c3cc861ff0fd35b0753a57cd21a7340fa3f22/test/runtests.jl#L38-L55
In the above snippet, `NodeData` is the struct type/datatype that we defined. 

Does that make sense?

Reply:
Maybe the part that is confusing is that types also have a type.  Maybe this helps:
```julia
julia> typeof(3.14)
Float64

julia> typeof(typeof(3.14))
DataType

julia> struct MyType
           a::Int
           b::Bool
       end

julia> instance = MyType(3, true)
MyType(3, true)

julia> instance.a, instance.b
(3, true)

julia> typeof(instance)
MyType

julia> typeof(MyType)
DataType
```

Reply:
Ah I see that the :: is specifying the expected datatype of the input. And I checked tests but missed the validate call.
Is it true that for required arguments you don't use "=" but for optional arguments you do?

For instance for this line, VSCode is pushing me to write it this way:
`validate_df(df, data_schema, fname = file_name, silent = silent)`

instead of this way:
`validate_df(df, schema = data_schema, file_name, silent)`

Reply:
This is actually slightly different, the reason VS code is suggesting the keyword argument form: `fname = file_name`, is because they are mandatory keyword arguments.  If you look at the definition of `validate_df`, it looks like this: `validate_df(df, schema; fname="", silent=false)`; the semicolon denotes that all following arguments are of the keyword argument form.